darkcluster/src/main/java/com/linkedin/darkcluster/api/DarkDispatcherVerifier.java

```diff
@@ -0,0 +1,20 @@
+package com.linkedin.darkcluster.api;
+
+import com.linkedin.r2.message.RequestContext;
+import com.linkedin.r2.message.rest.RestRequest;
+
+
+/**
+ * Interface that lets users define custom logic to determine if a given request is to be dispatched to dark cluster or not.
+ */
+@FunctionalInterface
+public interface DarkDispatcherVerifier
```

Comment:
this name can be confused with the other Verifier ... and you can't tell what's what when configuring config properties. What about DarkCustomGatekeeper? DarkGatekeeper?

Reply:
Will call it DarkGateKeeper